src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -751,8 +751,9 @@ DWORD InputStateMachineEngine::_GetGenericKeysModifierState(const VTParameters p
     //   HOME, END, PAGE UP, PAGE DOWN, and direction keys in the clusters to the left
     //   of the keypad; and the divide (/) and ENTER keys in the keypad.
     // This snippet detects the non-direction keys
-    const GenericKeyIdentifiers identifier = parameters.at(0);
-    if (identifier <= GenericKeyIdentifiers::Next)
+    const auto identifier = parameters.at(0).value();
+    if (identifier >= WI_EnumValue(GenericKeyIdentifiers::GenericHome) &&
+        identifier <= WI_EnumValue(GenericKeyIdentifiers::Next))
     {
         modifiers = WI_SetFlag(modifiers, ENHANCED_KEY);

```

Comment:
While we're in this file, it's worth noting that the assignment isn't actually needed here. This could just be:
```
WI_SetFlag(modifiers, ENHANCED_KEY);
```
There's another case like this on line 328.